### PR TITLE
feat: Add maven version comparison SQL function.

### PR DESCRIPTION
**Pull request by bobmcwhirter**

### CODE REVIEW

File: migration/tests/mavenver.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    if let Some(result) = result {
+        Ok(result.try_get_by_index(0)?)
+    } else {
+        Ok(None)
+    }
+}
+
+#[test_context(TrustifyContext)]
+#[test(tokio::test)]
+async fn test_mavenver_cmp(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
```
Comment (JimFuller-RedHat) [inline migration/tests/mavenver.rs:36]: have you considered any negative tests ... though maybe some error handling on version_matches might be needed

**Review by JimFuller-RedHat — APPROVED** [review]
all is well here

File: migration/tests/mavenver.rs (hunk 1 above)
Comment (bobmcwhirter) [reply migration/tests/mavenver.rs:36]: Such as comparing "1.0" to "tacos" ?

Comment (JimFuller-RedHat) [reply migration/tests/mavenver.rs:36]: ya valid input I guess ... though admittedly perhaps this is a guard rail best down higher up in version_matches

**Pull request merged**